cbmc/proofs/IotMqtt_SubscribeAsync/Makefile

```diff
@@ -0,0 +1,68 @@
+ABSTRACTIONS += --remove-function-body _IotMqtt_SerializeUnsubscribeCommon
+ABSTRACTIONS += --remove-function-body _IotMqtt_SerializeUnsubscribe
+ABSTRACTIONS += --remove-function-body _mqttOperation_match
+ABSTRACTIONS += --remove-function-body _mqttSubscription_setUnsubscribe
+ABSTRACTIONS += --remove-function-body _packetMatch
+ABSTRACTIONS += --remove-function-body _scheduleNextRetry
+ABSTRACTIONS += --remove-function-body _topicFilterMatch
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceClose
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceReceive
+
+OBJS += $(ENTRY)_harness.goto
+OBJS += $(MQTT)/cbmc/proofs/mqtt_state.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_api.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_helper.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_operation.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_serialize.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_subscription.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_validate.goto
+
+# One more than actual number of subscriptions in a subscription list
+SUBSCRIPTION_COUNT_MAX=2
+DEF += -DSUBSCRIPTION_COUNT_MAX=$(SUBSCRIPTION_COUNT_MAX)
+
+# One more than actual number of operations in an operation list
+OPERATION_COUNT_MAX=2
+DEF += -DOPERATION_COUNT_MAX=$(OPERATION_COUNT_MAX)
+
+# One more than actual length of topics
+TOPIC_LENGTH_MAX=3
+DEF += -DTOPIC_LENGTH_MAX=$(TOPIC_LENGTH_MAX)
+
+# Should be 2*SUBSCRIPTION_COUNT_MAX-1
+SUBSCRIPTION_LIST_MAX=3
+DEF += -DSUBSCRIPTION_LIST_MAX=$(SUBSCRIPTION_LIST_MAX)
+
+DEF += -DCBMC_STUB_REMOVEALLMATCHES=1
+
+# Allow MQTT allocations to fail for coverage
+DEF += -include mqtt_state.h
+DEF += -DIotMqtt_MallocMessage=malloc_can_fail
+DEF += -DIotMqtt_MallocOperation=malloc_can_fail
+DEF += -DIotMqtt_MallocSubscription=malloc_can_fail
+
+LOOP += _IotMqtt_AddSubscriptions.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += _IotMqtt_SerializeSubscribeCommon.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += _IotMqtt_SubscriptionPacketSize.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += _IotMqtt_ValidateSubscriptionList.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += _validateSubscription.0:$(TOPIC_LENGTH_MAX)
+LOOP += allocate_IotMqttSubscriptionArray.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += allocate_IotMqttSubscriptionList.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_FindFirstMatch.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_FindFirstMatch\$$link1.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_RemoveAllMatches.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link1.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += strncmp.0:$(TOPIC_LENGTH_MAX)
+LOOP += valid_IotMqttOperationList.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += valid_IotMqttSubscriptionArray.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += valid_IotMqttSubscriptionList.0:$(SUBSCRIPTION_COUNT_MAX)
+
+UNWINDING += --unwind 1
+UNWINDING += --unwindset '$(shell echo $(LOOP) | sed 's/ /,/g')'
```

Comment:
Should we move to avoiding the shell invocation?  (You advised me to stop in a different proof, to use multiple --unwindset, but I may not be consistent myself.)

Reply:
Yes, we should. We even have issue https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/812 to do a clean up the existing proofs. I've updated the Makefile to use multiple `--unwindset`, done.

Reply:
When I updated the Makefile with multiple `--unwindset` there were unwinding assertion failures, so I suggest to keep the old version and treat this on a separate PR for all proofs.

Reply:
There were unwinding assertions failures when I moved to `--unwindset`, so I returned to the old version. I'd suggest to update all Makefiles to use multiple `--unwindset` on a separate PR.